util/eventbus/bus.go

```diff
@@ -13,12 +13,29 @@ import (
 	"tailscale.com/util/set"
 )
 
+type publishedEvent struct {
+	Event     any
+	From      *Client
+	Published time.Time
+	Accepted  time.Time
+}
+
+type routedEvent struct {
+	Event     any
+	From      *Client
+	To        []*Client
+	Published time.Time
+	Accepted  time.Time
+	Routed    time.Time
+}
+
 // Bus is an event bus that distributes published events to interested
 // subscribers.
 type Bus struct {
-	router   *worker
-	write    chan publishedEvent
-	snapshot chan chan []publishedEvent
+	router     *worker
+	write      chan publishedEvent
+	snapshot   chan chan []publishedEvent
+	routeDebug hook[routedEvent]
```

Comment:
Idle terminological musing: Should we use "Trac(e|ing)" instead of "Debug(ging)?" ?

Reply:
I think I'd like to stick with the more generic "debug", because to me tracing conjures specific images of spans and opentelemetry and so forth, which this API can enable, but doesn't force.